packages/js/src/plugins/auctionHouseModule/updateAuctionHouse.ts

```diff
@@ -149,4 +159,50 @@ export const updateAuctionHouseBuilder = (
         key: params.instructionKey ?? 'updateAuctionHouse',
       })
     );
+
+  if (!auctionHouse.hasAuctioneer && params.auctioneerAuthority) {
+    const auctioneerAuthority = params.auctioneerAuthority
+      ?.publicKey as PublicKey;
+    const ahAuctioneerPda = findAuctioneerPda(
+      auctionHouse.address,
+      auctioneerAuthority
+    );
+
+    const ahAuctioneerAccount = await metaplex
+      .rpc()
+      .getAccount(ahAuctioneerPda);
+
+    // Skip delegate if it has been made already
+    if (ahAuctioneerAccount.exists) {
+      return builder;
+    }
+
+    const scopes = params.auctioneerScopes ?? [
+      AuthorityScope.Deposit,
+      AuthorityScope.Buy,
+      AuthorityScope.PublicBuy,
+      AuthorityScope.ExecuteSale,
+      AuthorityScope.Sell,
+      AuthorityScope.Cancel,
+      AuthorityScope.Withdraw,
+    ];
+
+    builder.add({
+      instruction: createDelegateAuctioneerInstruction(
```

Comment:
We need to abstract both **delegating to** and **updating** Auctioneer instances.

See https://github.com/metaplex-foundation/metaplex-program-library/blob/cac9cfd042cc5a16f50d635074555392d3b9044e/auction-house/js/src/generated/instructions/updateAuctioneer.ts#L66